### PR TITLE
added dialog closing on out-of-dialog clicks and esc key

### PR DESCRIPTION
**Pull request by Pomax**

### CODE REVIEW

**Comment by Pomax** [comment]
closes https://github.com/mozilla/webpagemaker/issues/268 when landed

File: js/fc/ui/modals.js [hunk 1]
```diff
@@ -7,6 +7,27 @@ define(function() {
...
+      $(".modal-overlay").hide();
+    }
+
+    /**
+     * When someone clicks on the darkening-overlay, rather
+     * than the modal dialog, close the modal dialog again.
+     */
+    $(".modal-overlay").click(function() { $(this).hide(); });
+    // intercept and block for clicks inside the modal
+    $(".modal-overlay div").click(function() { return false; });
```
Comment (brianloveswords) [inline js/fc/ui/modals.js]: You should use `event.stopPropagation();` here

**Comment by Pomax** [comment]
The problem is that .modal-overlay must be closed only if the event source is the overlay itself, not its content, so either there's an event.target==this for .modal-overlay elements, or the modal content should intercept. event.target is preferable so if IE supports that, I'll use that.

**Comment by Pomax** [comment]
amended, using event.target===this as sanity check now.

**Comment by brianloveswords** [comment]
Nice, that's a lot cleaner. :thumbsup:

**Pull request merged**